### PR TITLE
Document that Process::command will search the PATH

### PR DESCRIPTION
**Pull request by sourcefrog**

### CODE REVIEW

File: src/libstd/process.rs [hunk 1]
```diff
@@ -253,6 +253,10 @@ impl Command {
     /// Builder methods are provided to change these defaults and
     /// otherwise configure the process.
     ///
+    /// The `PATH` from the Command's environment will be searched
+    /// if the program contains only a single path component, as for
+    /// `execvp`.
```
Comment (nagisa) [inline src/libstd/process.rs]: I don’t get what “as for `execvp`“ part is supposed to mean. Why are you including references to POSIX specific functions in documentation of platform agnostic part of documentation?

Comment (sourcefrog) [inline src/libstd/process.rs]: Yeah I was wondering whether that would help make the behavior clearer to people or if it was better style not to mention it.  I can remove that phrase.

Comment (sfackler) [inline src/libstd/process.rs]: I would just give a couple of examples - "For example, `PATH` will be searched for `rustc` but not `bin/rustc`".

File: src/libstd/process.rs [hunk 2]
```diff
@@ -253,6 +253,10 @@ impl Command {
     /// Builder methods are provided to change these defaults and
     /// otherwise configure the process.
     ///
+    /// The `PATH` from the Command's environment will be searched
```
Comment (sourcefrog) [inline src/libstd/process.rs]: done

File: src/libstd/process.rs (hunk 1 above)
Comment (sourcefrog) [inline src/libstd/process.rs]: done

Comment (ollie27) [inline src/libstd/process.rs]: That isn't the case on Windows. It uses the child's `PATH` (badly: #37519) even if the program name contains multiple path components.

Windows and Linux have very different behaviour here.

Comment (sourcefrog) [inline src/libstd/process.rs]: Yes, I saw that, what a mess.

The main thing I found missing here, as a reader, is that unqualified names will be searched along the PATH.  (That's implied by some examples but I think worth having clear.)

I'd like to also say you can control that by setting the PATH environment, and it seems you can, except for various bugs on Windows. Perhaps this is too complex to get into here? It looks like general Rust style is not to mention bugs or caveats in the API docs?

Comment (ollie27) [inline src/libstd/process.rs]: >    If program is not an absolute path, the PATH will be searched in an OS-defined way.

Something like that seems reasonable. I think the main thing that we need to include is that the behaviour is OS dependant.

**Comment by alexcrichton** [comment]
@ollie27's [concerns](https://github.com/rust-lang/rust/pull/38018#discussion_r89679176) make sense to me, perhaps this could mention that the behavior is still platform-specific and basically just say that it's attempted to interpret `PATH` in the context of the child process, if required?

**Comment by sourcefrog** [comment]
How about this?

**Comment by alexcrichton** [comment]
Sounds reasonable to me! Looks like the tidy checks are failing though?

**Comment by sourcefrog** [comment]
whitespace fixed.

**Comment by alexcrichton** [comment]
Thanks! Could you also squash the commits?

**Comment by sourcefrog** [comment]
Done, commits squashed

**Pull request merged**